### ISSUE TITLE
add feature recognition method in findElementByImage

### ISSUE THREAD

**Issue by weikilla** [appium/appium#16331]
## The problem

The success rate of template matching is not very high on different resolution devices, hope to add  feature recognition method. reference to netease airtest automation framework: [link](https://airtest.doc.io.netease.com/tutorial/9_Improved_compatibility/#2_2)

**Comment by mykola-mokhnach** [comment]
Please provide a proper description for the feature. We don't understand Chinese

**Comment by jlipps** [comment]
Is there a feature recognition method in OpenCV that we could use?

**Comment by weikilla** [comment]
> 

sorry, see :https://airtest.doc.io.netease.com/en/IDEdocs/airtest_framework/1_script_settings/

**Comment by weikilla** [comment]
> 



> Is there a feature recognition method in OpenCV that we could use?

yes, for example: [surf](https://airtest.doc.io.netease.com/en/IDEdocs/airtest_framework/1_script_settings/#image-recognition-algorithm-cvstrategy)

**Comment by weikilla** [comment]
> Please provide a proper description for the feature. We don't understand Chinese

add description: https://github.com/AirtestProject/Airtest/blob/master/benchmark/README.md

**Comment by KazuCocoa** [comment]
Once we add `surf` in https://github.com/appium/appium-support/blob/15afcad2329e872f76195120ff238450fafc3bbb/lib/image-util.js#L41-L51 probably users can use them?

Then, probably we need to add settings api to configure the detection method for findByImage
https://github.com/appium/appium-plugins/blob/1cf7aef35a79371262bc8488a54efa564a004719/packages/images/lib/finder.js#L163

**Comment by mykola-mokhnach** [comment]
As @KazuCocoa mentioned above a couple of recognition methods are currently supported. Unfortunately we completely depend on the opencv4nodejs bindings there, which are not being maintained properly. We'd be happy if someone could help us to fix existing issues for the integration with that lib.

**Comment by mykola-mokhnach** [comment]
According to the information I found so far about the WASM binary used in opencv-bindings JS OpenCV wrapper.

Likely supported detectors:

AKAZE — whitelisted in OpenCV.js config 
KAZE — present in the whitelist 
ORB — very likely supported 
BRISK — whitelisted for create and getDefaultName 
MSER — whitelisted
FAST / Agast feature detectors — present via FastFeatureDetector and AgastFeatureDetector in the whitelist. 
GFTTDetector — in whitelist too. 
SimpleBlobDetector — whitelisted. 

Not supported (or not by default):

SIFT — despite being part of features2d in OpenCV C++, it’s not reliably exposed in OpenCV.js builds unless custom-built. 
SURF / other contrib / non-free detectors — unlikely, since they are typically part of the xfeatures2d or contrib modules and not whitelisted in the default JS build.
